examples/zephyr/common/Kconfig

```diff
@@ -13,6 +13,22 @@ if GOLIOTH_SAMPLE_COMMON
 
 rsource "Kconfig.defconfig"
 
+choice GOLIOTH_SAMPLE_AUTH_TYPE
```

Comment:
Should we make a default for this in https://github.com/golioth/golioth-firmware-sdk/blob/main/examples/zephyr/common/Kconfig.defconfig ?

Reply:
First options specified are default. Since PKI and PSK are exclusive based on `depends on GOLIOTH_AUTH_METHOD_PSK` and `depends on GOLIOTH_AUTH_METHOD_CERT` the only option that is non-default is newly introduced `GOLIOTH_SAMPLE_AUTH_TYPE_TAG`. So I don't thing we need to change anything.